orangecontrib/text/widgets/owloadcorpus.py

```diff
@@ -83,6 +84,7 @@ def open_file(self, path):
         if path:
             try:
                 self.corpus = Corpus.from_file(path)
+                self.corpus.name = os.path.basename(path)[:-4]
```

Comment:
What if someday we support .xlsx? I suggest to split by dot and take the first token. Or even better use _splitext_:

```py
>>> os.path.splitext(os.path.basename("/a/b/c.txt"))[0]
Out[1]: 'c'
```